File: 프로그래머스/0/181926. 수 조작하기 1/수 조작하기 1.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

int solution(int n, const char* control) {
    for (int i = 0; i < strlen(control); i++) {
        if (control[i] == 'w') n += 1;
        else if (control[i] == 's') n -= 1;
        else if (control[i] == 'd') n += 10;
        else if (control[i] == 'a') n -= 10;
    }
    return n;
}

File: 프로그래머스/0/181926. 수 조작하기 1/수 조작하기 1.js
const operations = {
    w: (n) => n + 1,
    s: (n) => n - 1,
    d: (n) => n + 10,
    a: (n) => n - 10,
};


function solution(n, control) {
    return [...control].reduce((prev, op) => operations[op](prev), n);
}